src/main/java/uk/gov/digital/ho/hocs/workflow/api/WorkflowService.java

```diff
@@ -357,4 +362,39 @@ public void updateStage(UUID caseUUID, UUID stageUUID, Map<String, String> value
 
         camundaClient.completeTask(stageUUID, values);
     }
+
+    public ResponseEntity closeCase(UUID caseUUID) throws JsonProcessingException {
+        GetCaseworkCaseDataResponse caseDetails = caseworkClient.getCase(caseUUID);
+
+        //Get stage uuid from casework
+        GetStagesResponse stage = caseworkClient.getActiveStage(caseDetails.getReference().replace("/", "%2F"));
```

Comment:
I think there's a URLEncode method lurking around that would do this properly..

Reply:
https://github.com/UKHomeOffice/hocs-frontend/blob/main/server/libs/encodingHelpers.js - we have this in the frontend but can't find anything similar in workflow.

Reply:
https://docs.oracle.com/javase/7/docs/api/java/net/URLEncoder.html